### DISCUSSION TITLE
Multiple configs for a single rule

### DISCUSSION THREAD

**Discussion by piranna** [eslint/eslint#15235] [Q&A] [answered]
Not sure if it's possible at all and would need it to be implemented...

At https://eslint.org/docs/user-guide/configuring/rules#configuring-rules it says that rules can be set to `off`, `warn` and `error`. For simple ones that's enough, but since more complex ones can be configured, how is it possible to set a rule to `warn` for one rule config and `error` for another one? For example, I'm interested on throw an error on `no-restricted-globals` rule when using `setInterval`, but just show a warning message when using `setTimeout`. How can I achieve this?

**Comment by aladdin-add** [comment, marked as answer]
eslint does not support this way. a workaround is to define a new rule (e.g. no-setTimeout), so you you can config:
```js
{
  rules: {
    'no-restricted-globals': 'error',
    'no-setTimeout': 'warn',
  }
}
```

**Reply by piranna** [reply]
And how can I define that rule?

**Reply by aladdin-add** [reply]
you can use this one: https://github.com/not-an-aardvark/eslint-rule-composer

**Reply by piranna** [reply]
Really interesting, thanks! :-D
